lib/chainweb/p2p/cut.ex

```diff
@@ -0,0 +1,29 @@
+defmodule Kadena.Chainweb.P2P.Cut do
+  @moduledoc """
+  Cut endpoints implementation for P2P API.
+  """
+
+  @endpoint "cut"
+
+  alias Kadena.Chainweb.P2P.CutResponse
+  alias Kadena.Chainweb.{Error, Request}
+
+  @type opts :: Keyword.t()
+  @type error :: {:error, Error.t()}
+  @type cut_response :: CutResponse.t() | error()
+
+  @spec retrieve(opts :: opts()) :: cut_response()
+  def retrieve(opts) do
+    location = Keyword.get(opts, :location)
+    network_id = Keyword.get(opts, :network_id, :testnet04)
+    query_params = Keyword.get(opts, :query_params, nil)
```

Comment:
@EdwinGuayacan some suggestions here!

1. We should set a default value for `opts`, it means `opts \\ []`.
So to make a request to `https://api.chainweb.com/chainweb/0.0/mainnet01/cut` the user only had to call:
    ```elixir
    Kadena.Chainweb.P2P.Cut.retrieve()
    ```

2. In line 19, use `Keyword.get/2` instead of sending `nil` as the third arg. https://hexdocs.pm/elixir/1.12/Keyword.html#get/3
3. IMO, a more descriptive approach to query params is:
    ```elixir
    Kadena.Chainweb.P2P.Cut.retrieve(query_params: [maxheight: 123456])
    ```
    In this way, it's clear we're setting a max height.